storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerConfig.java

```diff
@@ -227,9 +229,9 @@ public String toString() {
                 ", metadataTopicReplicationFactor=" + metadataTopicReplicationFactor +
                 ", initializationRetryMaxTimeoutMs=" + initializationRetryMaxTimeoutMs +
                 ", initializationRetryIntervalMs=" + initializationRetryIntervalMs +
-                ", commonProps=" + commonProps +
-                ", consumerProps=" + consumerProps +
-                ", producerProps=" + producerProps +
+                ", commonProps=" + configMapToRedactedString(commonProps, AdminClientConfig.configDef()) +
+                ", consumerProps=" + configMapToRedactedString(consumerProps, ConsumerConfig.configDef()) +
+                ", producerProps=" + configMapToRedactedString(producerProps, ProducerConfig.configDef()) +
```

Comment:
Please add a unit test to verify that no sensitive strings are output

Reply:
@chia7712 thanks for the review! Updated test to include validations